### ISSUE TITLE
`--cv` selects and evaluates on the same folds — `eval_*.csv` is not a held-out result

### ISSUE THREAD

**Issue by natalijamitic** [Alexasparis/AF3TCRpMHC#3]
Hi; I have been reading through the classification code with the aim of reusing the filter approach, and I think the cross-validation in `filters.py` is not doing what the README describes.

Apologies for the length; I wanted to include enough detail to make this easy to verify.

## What the code does

`create_nonoverlapping_folds` (`scripts_classification/filters.py:14`) correctly builds five `(train_df, test_df)` pairs, splitting on `pdb_id` so that no PDB appears on both sides of a fold.

However, both consumers appear to discard the test half:

```python
# filters.py:110, inside explore_filters
for train_df, _ in folds:

# filters.py:197, inside evaluate_filter
for train_df, _ in folds:
```

As a result:

1. `explore_filters` searches over metric/operator combinations using the training folds.
2. The selected filter is then passed to `evaluate_filter`.
3. `evaluate_filter` **scores it on those same training folds** again.

The `test_df` half of each fold tuple is never used in either path.

## What the README says

The README describes the output as:

> Evaluation CSV of the best filter on the test sets: `../filters/eval_<tiers_negative>_<tiers_positive>.csv`

and describes the workflow as:

> Explore filters using a 5 fold cross validation setting … and test it in the test set.

That sounds like `eval_*.csv` is intended to report performance on held-out fold data.

## Evidence from the committed outputs

For example:

```console
$ head -2 filters/eval_LQ_vs_AQ_MQ_HQ.csv
precision_mean,recall_mean,false_positive_rate_mean
0.9862770626760298,0.8805305194771724,0.14592264444142444
```

The corresponding top filter row reports the same metrics:

```text
precision_mean = 0.9862770626760298
recall_mean = 0.8805305194771724
false_positive_rate_mean = 0.14592264444142444
```

So the reported evaluation metrics are bit-identical to the metrics used for filter selection.

That is consistent with the current implementation: both values are computed from the training halves of the same folds rather than from held-out `test_df` data.

## Why it matters

The reported filter is selected from **265,720 candidate metric/operator combinations**:

```text
(3^12 - 1) / 2 = 265,720
```

and `filter_LQ_vs_AQ_MQ_HQ.csv` contains 265,721 lines including the header.

Selecting the best-performing candidate from such a large search space and then reporting its performance on the same data used for selection produces an **optimistically biased estimate of generalization performance**. The search can exploit fold-specific noise, so the resulting `eval_*.csv` values should not be interpreted as held-out test performance.

## Suggested fix

In `evaluate_filter`, I think the loop should use the held-out half:

```python
for _, test_df in folds:
```

with the rest of the evaluation performed on `test_df`.

`explore_filters` should continue using the training half; that part appears consistent with the intended selection/evaluation split.

The code change itself is small, but it will change the values written to `eval_*.csv`, so I thought it was better to raise this as an issue first rather than open a PR that silently changes a published result.

## One related detail in the same function

There is also a separate fold-construction detail that may be worth looking at.

`create_nonoverlapping_folds:35` stratifies using:

```python
pdb_quality = df.drop_duplicates('pdb_id')[['pdb_id', 'Quality']].copy()
```

Because `drop_duplicates` keeps the first row for each `pdb_id`, each PDB is assigned the quality tier of whichever model happens to appear first in the DataFrame.

If models belonging to the same PDB can span multiple quality tiers, then this does not necessarily balance the actual model-level tier distribution across folds; it balances only the first retained tier per PDB.

Also, this line:

```python
rng = np.random.default_rng(random_state)
```

appears to create an RNG that is never used, since the shuffle is performed through `DataFrame.sample(random_state=...)`. That is harmless, but slightly confusing when reading the fold-generation logic.

Happy to prepare a PR for any of this if useful. I mainly wanted to check that I am reading the intended CV procedure correctly before proposing a behavior-changing fix.
